### PR TITLE
cluster-clean, Fix wait for namespace to disappear logic

### PR DESCRIPTION
**Pull request by oshoval**
The logic was not correct, it waited for the wrong variable.
Adjust parameters and fix logic.

Signed-off-by: Or Shoval <oshoval@redhat.com>

```release-note
None
```

### CODE REVIEW

File: hack/cluster-clean.sh [hunk 1]
```diff
@@ -120,22 +120,25 @@ function delete_resources() {
...
+
+            local current_time=0
+            local sample=3
+            local timeout=180
+            echo "Waiting for ${ns} namespace to disappear ..."
+            set +x
+            while [ -n "$(_kubectl get ns | grep -w ${ns})" ]; do
                 sleep $sample
-                start_time=$((current_time + sample))
+                local current_time=$((current_time + sample))
```
Comment (rhrazdil) [inline hack/cluster-clean.sh:135]: I thik the `local` keyword is superfluous here, isn't it?

**Review by rhrazdil — COMMENTED** [review]
Thanks, one question

File: hack/cluster-clean.sh (hunk 1 above)
Comment (oshoval) [reply hack/cluster-clean.sh:135]: else the var is accessible out of the loop

Comment (rhrazdil) [reply hack/cluster-clean.sh:135]: It's already local though
```
function delete_namespaces() {
  local current_time=12
  current_time=42
  time=24
  local time=334
}

delete_namespaces

echo "current_time: $current_time"
echo "time: $time"
```
yields
```
 rhrazdil@rhrazdilt480s  ~  bash test.sh
current_time: 
time: 24
```

Comment (oshoval) [reply hack/cluster-clean.sh:135]: ah you right forgot it was declared before

**Comment by oshoval** [comment]
addressed comments

**Review by rhrazdil — APPROVED** [review]
Thanks!
/lgtm

**Review by RamLavi — APPROVED** [review]
/lgtm

File: hack/cluster-clean.sh [hunk 2]
```diff
@@ -120,22 +120,25 @@ function delete_resources() {
...
+    for ns in ${managed_namespaces[@]}; do
+        if [ -n "$(_kubectl get ns | grep "${ns} ")" ]; then
+            echo "Clean ${ns} namespace"
+            _kubectl delete ns ${ns}
+
+            local current_time=0
+            local sample=3
+            local timeout=180
+            echo "Waiting for ${ns} namespace to disappear ..."
+            set +x
```
Comment (rmohr) [inline hack/cluster-clean.sh:132]: Is the `set +x` a leftover from debugging?

**Review by rmohr — COMMENTED** [review]
One question.

File: hack/cluster-clean.sh (hunk 2 above)
Comment (oshoval) [reply hack/cluster-clean.sh:132]: nope, i return the set -x on line 141
else it will just print useless info each `sample` seconds

**Comment by oshoval** [comment]
/retest

`pull-kubevirt-e2e-k8s-1.20-sig-storage` lane is in a bad condition (not related to this PR)

**Pull request merged**